src/frontend/screens/Settings/ProjectSettings/YourTeam/WaitingForInviteAccept.tsx

```diff
@@ -0,0 +1,100 @@
+import {BackHandler, StyleSheet, View} from 'react-native';
+import InviteSent from '../../../../images/InviteSent.svg';
+import {Text} from '../../../../sharedComponents/Text';
+import {defineMessages, useIntl} from 'react-intl';
+import React from 'react';
+import {TextButton} from '../../../../sharedComponents/TextButton';
+import {useFocusEffect} from '@react-navigation/native';
+import {NativeRootNavigationProps} from '../../../../sharedTypes';
+import {useProject} from '../../../../hooks/server/projects';
+import {useQueryClient} from '@tanstack/react-query';
+import {ErrorModal} from '../../../../sharedComponents/ErrorModal';
+import {useBottomSheetModal} from '../../../../sharedComponents/BottomSheetModal';
+
+const m = defineMessages({
+  waitingMessage: {
+    id: 'screens.Setting.ProjectSettings.YourTeam.WaitingForInviteAccept.waitingMessage',
+    defaultMessage: 'Waiting for Device to Accept Invite',
+  },
+  timerMessage: {
+    id: 'screens.Setting.ProjectSettings.YourTeam.WaitingForInviteAccept.timerMessage',
+    defaultMessage: 'Invite sent {seconds}s ago',
+  },
+  cancelInvite: {
+    id: 'screens.Setting.ProjectSettings.YourTeam.WaitingForInviteAccept.cancelInvite',
+    defaultMessage: 'Cancel Invite',
+  },
+});
+
+export const WaitingForInviteAccept = ({
+  navigation,
+  route,
+}: NativeRootNavigationProps<'WaitingForInviteAccept'>) => {
+  const {formatMessage: t} = useIntl();
+  const [time, setTime] = React.useState(0);
+  const {openSheet, closeSheet, isOpen, sheetRef} = useBottomSheetModal({
+    openOnMount: false,
+  });
+  const project = useProject();
+  const queryClient = useQueryClient();
+
+  React.useEffect(() => {
+    project.$member
+      .invite(route.params.deviceId, {roleId: route.params.role})
+      .then(() => {
+        queryClient.invalidateQueries({queryKey: ['projectMembers']}),
+          navigation.navigate('InviteAccepted', route.params);
+      })
+      .catch(err => {
+        openSheet();
+      });
+  }, []);
```

Comment:
The architecture is a little odd here. But essentially, when the coordinator clicks to invite someone, they are navigated to this page to wait for the invite. So im deferring the actual invite till after the navigation has happened so we have access to the promise. Otherwise, the user clicks invite, the invite promise is created, and then the user is navigated to this page. Since react-nav doesnt allow for non-serializable things to be passed between pages, it would not have access to the promise.

Reply:
I updated this.

I dont think we need to worry about all 4 states in one screen. One screen just needs to handle sending the invite, and tracking the promise that comes along with it. As soon as the promise is resolved or rejected, we can just navigate to a new page. So iI[updated the page](https://github.com/digidem/CoMapeo-mobile/blob/4ceee0526952470d6cab3be8a92ec878c7fd1afd/src/frontend/screens/Settings/ProjectSettings/YourTeam/ReviewInvitation.tsx#L50) to do that.

Unfortunately we don't get the transition animation between sending and invite and waiting for an invite that react navigation gave us, but for now I think that is ok.